contracts/DependencyAudit.cdc

```diff
@@ -41,27 +41,99 @@ access(all) contract DependencyAudit {
         }
 
         if unstagedDependencies.length > 0 {
-            if DependencyAudit.panicOnUnstaged {
-                // If `panicOnUnstaged` is set to true, the transaction will panic if there are any unstaged dependencies
-                // the panic message will include the unstaged dependencies
-                var unstagedDependenciesString = ""
-                var numUnstagedDependencies = unstagedDependencies.length
-                var j = 0
-                while j < numUnstagedDependencies {
-                    if j > 0 {
-                        unstagedDependenciesString = unstagedDependenciesString.concat(", ")
-                    }
-                    unstagedDependenciesString = unstagedDependenciesString.concat(unstagedDependencies[j].toString())
-
-                    j = j + 1
-                }
-
-                // the transactions will fail with a message that looks like this: `error: panic: Found unstaged dependencies: A.2ceae959ed1a7e7a.MigrationContractStaging, A.2ceae959ed1a7e7a.DependencyAudit`
-                panic("This transaction is using dependencies not staged for Crescendo upgrade coming soon! Learn more: https://bit.ly/FLOWCRESCENDO. Dependencies not staged: ".concat(unstagedDependenciesString))
-            } else {
-                emit UnstagedDependencies(dependencies: unstagedDependencies)
+            self.maybePanicOnUnstagedDependencies(unstagedDependencies)
```

Comment:
I think I'm missing something - why do we want to randomly panic here? I would think we'd want clear expectations around whether unstaged dependencies panic or not.

Reply:
The business  requirement is that there would be a random chance the transaction would fail if it is missing dependencies. That random chance would start at 0% and increase to 100% just before the spork.

Reply:
I see. I still don't understand the rationale behind that product decision, but this isn't the place to discuss prod reqs so I'll just review based on that stated requirement